pkg/apiserver/webhook/secret_webhook.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	controllerclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	crdv1alpha1 "antrea.io/nephe/apis/crd/v1alpha1"
+)
+
+var (
+	errorMsg00001 = "failed to get CloudProviderAccount list"
+)
+
+// nolint:lll
+// +kubebuilder:webhook:verbs=update;delete,path=/validate-v1-secret,mutating=false,failurePolicy=fail,groups="",resources=secrets,versions=v1,name=vsecret.kb.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
+
+// SecretValidator is used to validate Secret.
+type SecretValidator struct {
+	Client  controllerclient.Client
+	Log     logr.Logger
+	decoder *admission.Decoder
+}
+
+// Handle handles admission requests for Secret.
+func (v *SecretValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	v.Log.V(1).Info("Received admission webhook", "req", req, "operation", req.Operation)
+	switch req.Operation {
+	case admissionv1.Create:
+		return v.validateCreate(req)
+	case admissionv1.Update:
+		return v.validateUpdate(req)
+	case admissionv1.Delete:
+		return v.validateDelete(req)
+	default:
+		v.Log.Info("Invalid admission webhook request")
```

Comment:
In case of default, you aren't returning error

Reply:
Should we deny this request?